### DISCUSSION TITLE
Infrastructure as Code / config.toml for production

### DISCUSSION THREAD

**Discussion by NickG-NZ** [supabase/supabase#22220] [Feature Requests]
Surprised I can't find any mention of this yet.
Would be really great if we could have something akin to the config.toml file for managing our production services. This should completely replace any interaction with the GUI.
This way we can record changes in version control, run commands based on the configuration in CI and have much more parity between development and production environments.

Our supabase folder might contain two files: config.toml.dev and config.toml.prod.
Email templates would also be able to be synced between our production and development environments. (I saw a discussion on this).

Would also be great to be able to export the config.toml.prod from the current project settings.

**Comment by Taylor123** [comment]
I'm looking for the same feature. Without infra as code, it's difficult to coordinate updates across multiple developer machines and staging/prod environments. Have had many false bug reports from developers that forget to re-enable realtime or other settings after resetting/migrating their DB.
